app/assets/javascripts/discourse/app/lib/popup.js

```diff
@@ -0,0 +1,158 @@
+import { iconHTML } from "discourse-common/lib/icon-library";
+import I18n from "I18n";
+import tippy from "tippy.js";
+
+const GLOBAL_POPUPS_KEY = "new_user_tips";
+const POPUP_KEYS = ["first-notification", "topic-timeline"];
+
+const instances = {};
+const queue = [];
+
+// Plugin used to implement actions of the two buttons
+const PopupPlugin = {
+  name: "popup",
+
+  fn(instance) {
+    return {
+      onCreate() {
+        instance.popper
+          .querySelector(".btn-primary")
+          .addEventListener("click", (event) => {
+            const { currentUser, id } = instance.props;
+            hidePopupForever(currentUser, id);
+            event.preventDefault();
+          });
+
+        instance.popper
+          .querySelector(".btn-flat")
```

Comment:
It would be good to have more descriptive classes here for the "dismiss this popup" and "don't show me any popups" buttons. Relying on the button classes is a little too fragile here (for example, a theme could add "btn-flat" to the first button and that would cause problems with the events).

Reply:
Also, is there a way to clean up the click event listeners added here?

Reply:
I added more descriptive classes. I don't think it is necessary to clean up the click event listeners because the element is destroyed anyway and the browser will garbage collect the event handlers.